pkg/config/config.go

```diff
@@ -825,6 +825,16 @@ func (c *Config) ToFile(path string) error {
 	return os.WriteFile(path, b, 0o644)
 }
 
+// LogConfig pretty prints current config
+func (c *Config) LogConfig() error {
+	tomlConfig, err := c.ToBytes()
```

Comment:
There is an interesting aesthetic side-effect to using this approach—it does work, but when there is a list type value, which also so happens to be some levels of nesting deep from the top-level values, then the code that formats the output behind the scenes does a horrendous job with such list values. An example:

```toml
(...)
    [crio.runtime.runtimes]
      [crio.runtime.runtimes.runc]
        runtime_config_path = ""
        runtime_path = "/usr/sbin/runc"
        runtime_type = "oci"
        runtime_root = "/run/runc"
        allowed_annotations = ["io.containers.trace-syscall", "io.kubernetes.cri-o.Devices"]
        DisallowedAnnotations = ["io.kubernetes.cri-o.UnifiedCgroup", "io.kubernetes.cri-o.ShmSize", "io.kubernetes.cri.rdt-class", "org.systemd.property.", "org.criu.config", "io.kubernetes.cri-o.LinkLogs", "io.kubernetes.cri-o.DisableFIPS", "cpu-freq-governor.crio.io", "io.kubernetes.cri-o.seccompNotifierAction", "cpu-shared.crio.io", "module.wasm.image/variant", "io.kubernetes.cri-o.TrySkipVolumeSELinuxLabel", "io.kubernetes.cri-o.umask", "seccomp-profile.kubernetes.cri-o.io", "io.kubernetes.cri-o.PodLinuxResources", "io.kubernetes.cri-o.userns-mode", "cpu-quota.crio.io", "irq-load-balancing.crio.io", "cpu-c-states.crio.io", "io.kubernetes.cri-o.PodLinuxOverhead", "io.kubernetes.cri-o.cgroup2-mount-hierarchy-rw", "bundle", "cpu-load-balancing.crio.io", "io.kubernetes.cri.container-type", "run.oci."]
        monitor_path = "/usr/local/bin/conmon"
        monitor_cgroup = "system.slice"
        monitor_env = ["PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"]
        container_min_memory = "12MiB"
```

Looks so-so, isn't it? :smile: I haven't looked to deep into this. Perhaps there is a way to supply a custom formatted, or something like that.

That said, we might have a bug somewhere, per the above example output:

```toml
DisallowedAnnotations = ["io.kubernetes.cri-o.UnifiedCgroup", "io.kubernetes.cri-o.ShmSize", "io.kubernetes.cri.rdt-class", "org.systemd.property.", "org.criu.config", "io.kubernetes.cri-o.LinkLogs", "io.kubernetes.cri-o.DisableFIPS", "cpu-freq-governor.crio.io", "io.kubernetes.cri-o.seccompNotifierAction", "cpu-shared.crio.io", "module.wasm.image/variant", "io.kubernetes.cri-o.TrySkipVolumeSELinuxLabel", "io.kubernetes.cri-o.umask", "seccomp-profile.kubernetes.cri-o.io", "io.kubernetes.cri-o.PodLinuxResources", "io.kubernetes.cri-o.userns-mode", "cpu-quota.crio.io", "irq-load-balancing.crio.io", "cpu-c-states.crio.io", "io.kubernetes.cri-o.PodLinuxOverhead", "io.kubernetes.cri-o.cgroup2-mount-hierarchy-rw", "bundle", "cpu-load-balancing.crio.io", "io.kubernetes.cri.container-type", "run.oci."]
````

I would expect the name of this property (attribute) to be `disallowed_annotations`. I wonder if we are missing some formatting tag somewhere? So a small bug? Possibly?

Reply:
that value is not a configuration field, but is a public member of the config structure. I think we could change its public status but would need to update getters and setters I think

Reply:
@LenkaSeg, what do you think we could do here?

Reply:
The cleanest approach: getter + setter. The simplest, a TOML tag `toml:"-"`.

Reply:
Might be here?

https://github.com/cri-o/cri-o/blob/64fbde3a89741ca4fd64353e43cbf49f36c35a76/pkg/config/config.go#L224

https://github.com/cri-o/cri-o/blob/64fbde3a89741ca4fd64353e43cbf49f36c35a76/pkg/config/config.go#L227